.github/workflows/emscripten.yml

```
@@ -0,0 +1,35 @@
+name: Update Emscripten
+
+on:
+  pull_request:
+    types: [opened, synchronize]
+
+permissions:
+  contents: write
+  pull-requests: read
+
+jobs:
+  update-emscripten:
+    if: github.actor == 'dependabot[bot]'
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - name: Set up stable Rust toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@v1
+
+      - name: Run emscripten update xtask
+        run: cargo xtask update-emscripten
+
+      - name: Commit updated version
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git diff --exit-code cli/loader/emscripten-version || {
+            git remote set-url origin https://x-access-token:${{ secrets.GITHUB_TOKEN }}@github.com/$GITHUB_REPOSITORY
+            git add cli/loader/emscripten-version
+            git commit -m "build(deps): bump emscripten version"
+            git push origin HEAD:${{ github.event.pull_request.head.ref }}
+          }
```

Perhaps have the xtask create the commit like the wasmtime one does?
```suggestion
        env:
          GIT_AUTHOR_NAME: dependabot[bot]
          GIT_AUTHOR_EMAIL: 49699333+dependabot[bot]@users.noreply.github.com
          GIT_COMMITTER_NAME: dependabot[bot]
          GIT_COMMITTER_EMAIL: 49699333+dependabot[bot]@users.noreply.github.com
```